src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4104,38 +4143,61 @@ WorkerLimitCount(MultiExtendedOp *originalOpNode)
 	 */
 	if (groupClauseList == NIL)
 	{
-		workerLimitCount = originalOpNode->limitCount;
+		canPushDownLimit = true;
 	}
-	else if (sortClauseList != NIL)
+	else if (sortClauseList == NIL)
 	{
-		bool orderByNonAggregates = !(HasOrderByAggregate(sortClauseList, targetList));
-		bool canApproximate = CanPushDownLimitApproximate(sortClauseList, targetList);
+		canPushDownLimit = false;
+	}
+	else if (!hasOrderByAggregate)
+	{
+		canPushDownLimit = true;
+	}
+	else
+	{
+		canApproximate = CanPushDownLimitApproximate(sortClauseList, targetList);
+	}
 
-		if (orderByNonAggregates)
-		{
-			workerLimitCount = originalOpNode->limitCount;
-		}
-		else if (canApproximate)
-		{
-			Const *workerLimitConst = (Const *) copyObject(originalOpNode->limitCount);
-			int64 workerLimitCountInt64 = (int64) LimitClauseRowFetchCount;
-			workerLimitConst->constvalue = Int64GetDatum(workerLimitCountInt64);
+	/* create the workerLimitNode according to the made decision */
```

Comment:
Minor. This does not sound verbally right. Maybe 
..... according to the decision above